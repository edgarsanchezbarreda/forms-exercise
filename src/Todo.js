import React from 'react';

export const Todo = ({ task, id = '1', removeTodo }) => {
    const handleDeleteTodo = () => removeTodo(id);

    return (
        <div className='Todo'>
            <li>{task}</li>
            <button onClick={handleDeleteTodo}>X</button>
        </div>
    );
};
