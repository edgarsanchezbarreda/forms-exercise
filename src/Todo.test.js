import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Todo } from './Todo';

it('Should render without crashing', () => {
    render(<Todo />);
});

it('Should match snapshot', () => {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

it('Runs handleDelete function on click', () => {
    const removeTodoMock = jest.fn();
    const { getByText } = render(<Todo removeTodo={removeTodoMock} />);
    const removeButton = getByText('X');

    fireEvent.click(removeButton);
    expect(removeTodoMock).toHaveBeenCalled();
});
