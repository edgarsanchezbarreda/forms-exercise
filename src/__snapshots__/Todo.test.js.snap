// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <li />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
